zerver/models.py

```diff
@@ -1682,12 +1681,11 @@ def to_dict(self):
     @staticmethod
     def status_from_string(status):
         # type: (NonBinaryStr) -> Optional[int]
+        status_val = None
         if status == 'active':
             status_val = UserPresence.ACTIVE
         elif status == 'idle':
             status_val = UserPresence.IDLE
-        else:
-            status_val = None
```

Comment:
I don't think this 3-clause if statement going to a 2-clause one makes the code more readable.  Would either annotate as optional and/or report a mypy bug that it should be able to handle this sort of situation and tag the block as `Optional[foo]`..